### PR TITLE
Register form

### PR DESCRIPTION
**Pull request by sachabinder**
Working register form that allow to create teacher user.
This form has the flowing feature that should be test :
* Display error when trying to submit but one field is not set 
* Display error when trying to submit but one field has not correct format (see Regex in the RegisterForm component for more info)
* Display error on the top when the user already exists (communication with the Backend)

Fix some front issues such as adaptation of container size (because of the error display).
Renaming of some files to respect React conventions.

### CODE REVIEW

**Review by SullivanCastro — DISMISSED** [review]
Always a good job !

File: front/src/components/RegisterForm.js [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    if (!fields['firstName']) {
+      formIsValid = false;
+      errors['firstName'] = '*Merci d\'entrer un prénom';
+    }
+
+    if (typeof fields['firstName'] !== 'undefined') {
+      if (!fields['firstName'].match(/^[a-zA-Z-]+$/)) {
+        formIsValid = false;
+        errors['firstName'] =
+          '*Merci d\'utiliser uniquement que des caractères alphabétiques';
```
Comment (StephaneGrabsia) [inline front/src/components/RegisterForm.js]: Wouldn't have had "que" because it's redundant

File: front/src/components/RegisterForm.js [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    if (!fields['lastName']) {
+      formIsValid = false;
+      errors['lastName'] = '*Merci d\'entrer un nom';
+    }
+
+    if (typeof fields['lastName'] !== 'undefined') {
+      if (!fields['lastName'].match(/^[a-zA-Z-]+$/)) {
+        formIsValid = false;
+        errors['lastName'] =
+          '*Merci d\'utiliser uniquement que des caractères alphabétiques';
```
Comment (StephaneGrabsia) [inline front/src/components/RegisterForm.js]: Same

**Review by StephaneGrabsia — COMMENTED** [review]
Very good job, CR done

File: front/src/components/RegisterForm.js (hunk 1 above)
Comment (sachabinder) [reply front/src/components/RegisterForm.js]: sorry for that

**Review by StephaneGrabsia — APPROVED** [review]

**Pull request merged**